### ISSUE TITLE
Enable creating a new predicate from an existing one

### ISSUE THREAD

**Issue by nedsalk** [FuelLabs/fuels-ts#3392]
We should perhaps have a way of creating a new predicate from an existing one:
```ts
const loaderPredicate = await waitForDeploy();

const newLoaderPredicate = loaderPredicate.toNewInstance({ configurableConstants, data });
```
Because we currently don't have a nice way of duplicating a predicate but with new configurable constants and data.

_Originally posted by @nedsalk in https://github.com/FuelLabs/fuels-ts/pull/3389#discussion_r1838694868_

**Comment by YaTut1901** [comment]
@petertonysmith94 may I take this one? I am interested in smth more domain-related

**Comment by petertonysmith94** [comment]
@YaTut1901 yes you may take this one. It'll be a good one to get a better understanding of the domain, pop a message if you need anything :)
